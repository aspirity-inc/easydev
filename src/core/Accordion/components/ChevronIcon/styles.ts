import { css, styled } from 'styled-components';

import { Box } from '@core/Box';

import type { AnimationIconProps } from '../../types';

export const StyledChevronIcon = styled('span')`
  display: block;
  transition: color ${({ theme }) => theme.transition.default};
  color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['700'] : theme.colors.surface['300'])};
`;

export const AnimationWrapper = styled(Box)<AnimationIconProps>`
  transition: transform ${({ theme }) => theme.transition.default};

  ${({ $opened }) =>
    $opened &&
    css`
      transform: rotate(180deg);
    `}
`;
