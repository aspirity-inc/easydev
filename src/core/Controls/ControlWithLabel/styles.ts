import styled, { css } from 'styled-components';

type LabelProps = {
  reversed?: boolean;
};

export const StyledLabel = styled('label')<LabelProps>`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  gap: 8px;
  cursor: pointer;
  max-width: fit-content;

  ${({ reversed }) => {
    return (
      reversed &&
      css`
        flex-direction: row-reverse;
        justify-content: space-between;
      `
    );
  }}
`;
