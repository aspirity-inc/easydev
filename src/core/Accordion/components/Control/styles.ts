import { css, styled } from 'styled-components';

import { Box } from '@core/Box';
import { getTextVariants } from '@core/Typography/Text/styles';

import type { StyledControlProps } from '../../types';
import { StyledChevronIcon } from '../ChevronIcon/styles';

export const StyledTitle = styled(Box)`
  ${getTextVariants('body1')};
  color: inherit;
  transition: color ${({ theme }) => theme.transition.default};
`;

export const StyledSubtitle = styled(Box)`
  ${getTextVariants('body3')};
  font-family: inherit;
  line-height: 1.2;
  font-weight: 186;
  color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['700'] : theme.colors.surface['100'])};
`;

export const StyledIcon = styled(Box)`
  transition: color ${({ theme }) => theme.transition.default};
  color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['700'] : theme.colors.surface['300'])};
`;

export const StyledControl = styled('button')<StyledControlProps>`
  width: 100%;
  padding: 16px;
  margin: 0;
  border: none;
  outline: none;

  ${getTextVariants('body2')};
  font-family: inherit;
  line-height: 1.2;
  color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['900'] : theme.colors.surface['50'])};

  background-color: ${({ theme }) =>
    theme.type === 'light' ? theme.colors.surface['50'] : theme.colors.surface['900']};
  transition: background-color ${({ theme }) => theme.transition.default},
    color ${({ theme }) => theme.transition.default};
  cursor: pointer;

  ${({ $variant, $opened }) => ($variant === 'filled' ? getFilledStyles($opened) : getBorderedStyles($opened))}

  &:disabled {
    color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['500'] : theme.colors.surface['300'])};
    cursor: default;

    & ${StyledSubtitle} {
      color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['500'] : theme.colors.surface['300'])};
    }

    & ${StyledIcon} {
      color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['400'] : theme.colors.surface['600'])};
    }

    & ${StyledChevronIcon} {
      color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['500'] : theme.colors.surface['700'])};
    }
  }
`;

const getFilledStyles = ($opened: boolean) => {
  return css`
    &:hover:not([disabled]) {
      background-color: ${({ theme }) =>
        theme.type === 'light' ? theme.colors.surface['300'] : theme.colors.surface['800']};
    }

    &:focus-visible:not([disabled]),
    &:active:not([disabled]),
    &:focus:not([disabled]) {
      outline: none;
    }

    ${$opened &&
    css`
      background-color: ${({ theme }) =>
        theme.type === 'light' ? theme.colors.surface['300'] : theme.colors.surface['800']};
    `};
  `;
};

const getBorderedStyles = ($opened: boolean) => {
  return css`
    &:hover:not([disabled]) {
      & ${StyledIcon} {
        color: ${({ theme }) => (theme.type === 'light' ? theme.colors.primary['800'] : theme.colors.primary['500'])};
      }

      & ${StyledTitle} {
        color: ${({ theme }) => (theme.type === 'light' ? theme.colors.primary['800'] : theme.colors.primary['400'])};
      }

      & ${StyledChevronIcon} {
        color: ${({ theme }) => (theme.type === 'light' ? theme.colors.primary['700'] : theme.colors.primary['500'])};
      }
    }

    &:focus-visible:not([disabled]),
    &:active:not([disabled]),
    &:focus:not([disabled]) {
      outline: none;
    }

    ${$opened &&
    css`
      & ${StyledChevronIcon} {
        color: ${({ theme }) => (theme.type === 'light' ? theme.colors.primary['700'] : theme.colors.primary['500'])};
      }
    `};
  `;
};
