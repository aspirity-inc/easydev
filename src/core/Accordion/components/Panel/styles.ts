import { css, styled } from 'styled-components';

import { Box } from '@core/Box';
import { getTextVariants } from '@core/Typography/Text/styles';

import type { AnimationPanelProps, StyledControlProps } from '../../types';

export const AnimationWrapper = styled(Box)<AnimationPanelProps>`
  --duration: ${({ $duration }) => $duration}ms;

  height: 0;
  overflow: hidden;
  transition: height var(--duration) ease;
  border-bottom: 1px solid
    ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['300'] : theme.colors.surface['600'])};

  ${({ $opened, $height }) =>
    $opened &&
    css`
      height: ${$height}px;
    `}
`;

export const StyledPanel = styled(Box)<StyledControlProps>`
  height: auto;
  box-sizing: border-box;
  padding: 4px 16px 16px 16px;

  ${getTextVariants('body3')};
  font-family: inherit;
  line-height: 1.2;
  font-weight: 186;
  color: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['900'] : theme.colors.surface['50'])};

  ${({ $variant }) => ($variant === 'filled' ? getFilledStyles() : getBorderedStyles())}
`;

const getFilledStyles = () => {
  return css`
    background-color: ${({ theme }) =>
      theme.type === 'light' ? theme.colors.surface['300'] : theme.colors.surface['800']};
  `;
};

const getBorderedStyles = () => {
  return css`
    background-color: ${({ theme }) =>
      theme.type === 'light' ? theme.colors.surface['50'] : theme.colors.surface['900']};
  `;
};
