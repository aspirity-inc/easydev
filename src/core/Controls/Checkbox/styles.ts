import { styled } from 'styled-components';

import { ControlWrap, StyledInnerBase, StyledInput } from '@core/Controls/styles.ts';

export const StyledCheckbox = styled(StyledInput)``;

export const StyledCheckboxInner = styled(StyledInnerBase)``;

export const CheckboxWrap = styled(ControlWrap)`
  .check-mark {
    font-size: 22px;
  }
`;
