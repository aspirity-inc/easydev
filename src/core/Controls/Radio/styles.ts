import styled, { css } from 'styled-components';
import { ControlWrap, getControlColor, StyledInnerBase, StyledInput } from '@core/Controls/styles.ts';

export const StyledRadio = styled(StyledInput)``;

export const StyledRadioInner = styled(StyledInnerBase)`
  width: 16px;
  height: 16px;
  top: 50%;
  left: 50%;
  border-radius: 50%;
  transform: translate(-50%, -50%);
`;

export const RadioWrap = styled(ControlWrap)`
  border-radius: 50%;

  ${StyledRadioInner} {
    background-color: ${({ theme, color }) => getControlColor(theme, color) || theme.colors.tretiary['700']};
  }

  ${({ disabled }) =>
    disabled &&
    css`
      ${StyledRadioInner} {
        background-color: ${({ theme }) => theme.colors.surface['400']};
      }
    `}
`;
