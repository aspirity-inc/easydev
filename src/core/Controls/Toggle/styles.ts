import { css, styled } from 'styled-components';

// import DayBackgroundImage from './assets/toggle/day.svg';
// import NightBackgroundImage from './assets/toggle/night.svg';
import { ControlWrap, StyledInnerBase, getControlColor, StyledInput } from '../styles.ts';
import { ToggleWrapProps } from '../types.ts';

export const StyledToggle = styled(StyledInput)``;

export const StyledToggleInner = styled(StyledInnerBase)`
  width: 16px;
  height: 16px;
  border-radius: 50%;
  top: 50%;
  left: 4px;
  translate: 0 -50%;
  opacity: 1;
  transition: translate ${({ theme }) => theme.transition.default};
`;

export const ToggleWrap = styled(ControlWrap)<ToggleWrapProps>`
  --toggleBg: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['500'] : theme.colors.surface['200'])};
  --checkedBg: ${({ theme }) => (theme.type === 'light' ? theme.colors.tretiary['600'] : theme.colors.tretiary['400'])};
  --innerColor: ${({ theme }) => (theme.type === 'light' ? theme.colors.surface['50'] : theme.colors.surface['800'])};

  width: 40px;
  border-radius: 20px;
  background-color: var(--toggleBg);
  border: none;
  transition: background-color ${({ theme }) => theme.transition.default};

  ${StyledToggleInner} {
    background-color: var(--innerColor);
  }

  &:has(${StyledToggle}:checked) {
    background-color: ${({ theme, $color }) => getControlColor(theme, $color)};

    ${({ disabled }) =>
      disabled &&
      css`
        background-color: ${({ theme }) => theme.colors.surface['400']};
      `}
  }

  ${StyledToggle} {
    &:checked ~ ${StyledToggleInner} {
      translate: 100% -50%;
    }
  }

  ${({ disabled }) =>
    disabled &&
    css`
      background-color: ${({ theme }) =>
        theme.type === 'light' ? theme.colors.surface['400'] : theme.colors.surface['500']};

      ${StyledToggleInner} {
        filter: grayscale(1);
      }
    `}

  ${({ $isDayNightMode }) =>
    $isDayNightMode &&
    css`
      width: 48px;
      transition: background ${({ theme }) => theme.transition.default};

      ${StyledToggle} {
        transition: background ${({ theme }) => theme.transition.default};

        &:checked ~ ${StyledToggleInner} {
          translate: 150% -50%;
        }
      }
    `}

  ${({ $isDayNightMode, $isChecked }) => {
    console.log($isDayNightMode, $isChecked);
    return (
      $isDayNightMode &&
      ($isChecked
        ? css`
            background: url('/assets/toggle/day.svg');

            ${StyledToggleInner} {
              background: #ffc700;
            }
          `
        : css`
            background: url('/assets/toggle/night.svg');

            ${StyledToggleInner} {
              background: url('/assets/toggle/night-span.svg');
            }
          `)
    );
  }}
`;
