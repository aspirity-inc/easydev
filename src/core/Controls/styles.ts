import { css, type DefaultTheme, styled } from 'styled-components';

import { Box } from '@core/Box';
import { getTextVariants } from '@core/Typography/Text/styles.ts';

import type { ControlColorType, ControlContainerProps, ControlProps } from './types.ts';

export const StyledInput = styled('input')`
  margin: 0;
  padding: 0;

  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: inherit;
  z-index: 1;
`;

export const StyledInnerBase = styled('span')`
  position: absolute;
  left: -1px;
  top: -2px;
  color: inherit;
  opacity: 0;
  transition: opacity ${({ theme }) => theme.transition.default};
  z-index: 0;
`;

export const ControlWrap = styled('span')<ControlProps>`
  display: inline-block;
  position: relative;
  width: 24px;
  height: 24px;
  color: ${({ theme, $color }) => getControlColor(theme, $color)};
  background-color: ${({ theme }) => (theme.type === 'light' ? theme.colors.background : theme.colors.surface['800'])};
  border: 2px solid ${({ theme, $color }) => getControlColor(theme, $color)};
  border-radius: ${({ theme }) => theme.spacing.borderRadius.small}px;
  cursor: pointer;
  z-index: 2;

  ${StyledInput} {
    &:checked ~ ${StyledInnerBase} {
      opacity: 1;
    }
  }

  ${({ disabled }) =>
    disabled &&
    css`
      cursor: default;
      pointer-events: none;
      border-color: ${({ theme }) => theme.colors.surface['400']};
      color: ${({ theme }) => theme.colors.surface['400']};
    `}
`;

export const ControlContainer = styled('div')<ControlContainerProps>`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  gap: 8px;

  ${({ $reversed }) => {
    return (
      $reversed &&
      css`
        justify-content: flex-end;
        flex-direction: row-reverse;
      `
    );
  }}
`;

export const LabelContent = styled(Box)`
  cursor: pointer;
  user-select: none;
  ${getTextVariants('body3')}
`;

export const Label = styled(Box)``;

export const getControlColor = (theme: DefaultTheme, color?: ControlColorType) => {
  if (!color) return theme.type === 'light' ? theme.colors.tretiary['700'] : theme.colors.tretiary['400'];

  return theme.colors[color][500];
};
