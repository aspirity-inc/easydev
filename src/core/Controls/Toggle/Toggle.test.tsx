import { useState, useEffect } from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { expect, test, vi } from 'vitest';

import { EasydevProvider, THEMES } from '@core/Theme';
import { lightPalette } from '@core/Theme/themePalette';

import { Toggle } from '.';
import { StyledToggleInner } from './styles';
import { getControlColor } from '../styles';

test('toggle', async () => {
  const mockFn = vi.fn();
  const label = 'toggle';
  const { container } = render(
    <EasydevProvider>
      <Toggle defaultChecked onChange={mockFn} color="error" label={label} />
    </EasydevProvider>
  );

  const inputElement = container.querySelector('input') as Element;
  expect(inputElement).toHaveAttribute('checked');

  await userEvent.click(inputElement);
  expect(mockFn).toBeCalledTimes(1);

  expect(container.getElementsByClassName('easy_toggle-wrap')[0]).toHaveStyleRule(
    'color',
    getControlColor(THEMES.light, 'error')
  );

  expect(screen.getByText(label)).toBeInTheDocument();
});

test('disabled', () => {
  const { container } = render(
    <EasydevProvider>
      <Toggle disabled />
    </EasydevProvider>
  );

  const spanContainer = container.querySelector('span');
  expect(spanContainer).toHaveStyleRule('cursor', 'default');
  expect(spanContainer).toHaveStyleRule('pointer-events', 'none');
  expect(spanContainer).toHaveStyleRule('border-color', lightPalette.surface['400']);
  expect(spanContainer).toHaveStyleRule('color', lightPalette.surface['400']);

  expect(container.getElementsByClassName('easy_toggle-wrap')[0]).toHaveStyleRule(
    'background-color',
    lightPalette.surface['400']
  );
});

test('toggleBackground, toggleInnerBackground', async () => {
  const dayToggleBackground = "url('/assets/toggle/day.svg')";
  const nightToggleBackground = "url('/assets/toggle/night.svg')";

  const dayToggleInnerBackground = '#ffc700';
  const nightToggleInnerBackground = "url('/assets/toggle/night-span.svg')";

  const Component = () => {
    const [checked, setChecked] = useState(false);

    const [toggleBackground, setToggleBackground] = useState(checked ? dayToggleBackground : nightToggleBackground);
    const [toggleInnerBackground, setToggleInnerBackground] = useState(
      checked ? dayToggleInnerBackground : nightToggleInnerBackground
    );

    const onChange = () => setChecked((prevState) => !prevState);

    useEffect(() => {
      if (checked) {
        setToggleBackground(dayToggleBackground);
        setToggleInnerBackground(dayToggleInnerBackground);
      } else {
        setToggleBackground(nightToggleBackground);
        setToggleInnerBackground(nightToggleInnerBackground);
      }
    }, [checked]);

    return (
      <Toggle
        checked={checked}
        onChange={onChange}
        toggleBackground={toggleBackground}
        toggleInnerBackground={toggleInnerBackground}
        toggleWrapProps={{ style: { width: 48 } }}
        toggleInnerProps={{ style: { translate: checked ? '150% -50%' : '' } }}
      />
    );
  };
  const { container } = render(
    <EasydevProvider>
      <Component />
    </EasydevProvider>
  );

  const toggleWrapContainer = container.querySelector('.easy_toggle-wrap');
  expect(toggleWrapContainer).toHaveStyle('width: 48px');
  expect(toggleWrapContainer).toHaveStyleRule('background', nightToggleBackground);

  expect(toggleWrapContainer).toHaveStyleRule('background', nightToggleInnerBackground, {
    modifier: StyledToggleInner.toString(),
  });

  await userEvent.click(container.querySelector('input') as Element);
  expect(toggleWrapContainer).toHaveStyleRule('background', dayToggleBackground);
	expect(toggleWrapContainer).toHaveStyleRule('background', dayToggleInnerBackground, {
    modifier: StyledToggleInner.toString(),
  });
});
