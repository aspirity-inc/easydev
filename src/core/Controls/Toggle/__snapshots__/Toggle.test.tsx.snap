// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toggle 1`] = `
.c3 {
  margin: 0;
  padding: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 0;
  cursor: inherit;
  z-index: 1;
}

.c6 {
  position: absolute;
  left: -1px;
  top: -2px;
  color: inherit;
  opacity: 0;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  z-index: 0;
}

.c0 {
  display: inline-block;
  position: relative;
  width: 24px;
  height: 24px;
  color: #DE0000;
  background-color: #FFFFFF;
  border: 2px solid #DE0000;
  border-radius: 4px;
  cursor: pointer;
  z-index: 2;
}

.c0 .c2:checked~.c4 {
  opacity: 1;
}

.c7 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  top: 50%;
  left: 4px;
  -webkit-translate: 0 -50%;
  -moz-translate: 0 -50%;
  -ms-translate: 0 -50%;
  translate: 0 -50%;
  opacity: 1;
  -webkit-transition: translate 0.3s;
  transition: translate 0.3s;
}

.c8 .c5 {
  background-color: var(--innerColor);
}

.c8 .c5 {
  -webkit-translate: 100% -50%;
  -moz-translate: 100% -50%;
  -ms-translate: 100% -50%;
  translate: 100% -50%;
}

.c1 {
  --toggleBg: #A5A5A7;
  --checkedBg: #B693FF;
  --innerColor: #FFFFFF;
  width: 40px;
  border-radius: 20px;
  background-color: var(--toggleBg);
  border: none;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
}

.c1 .c5 {
  background-color: var(--innerColor);
}

<span
  class="c0 c1 easy_toggle-wrap"
>
  <input
    checked=""
    class="c2 c3 easy_toggle-item"
    id=":r0:"
    type="checkbox"
  />
  <span
    class="c4 c5 c6 c7 easy_toggle-inner"
  />
</span>
`;
